### ISSUE TITLE
Alacritty doesn't display emoji

### ISSUE THREAD

**Issue by pikajude** [jwilm/alacritty#153]
I know, I know.

Anyway, iTerm 2 and Terminal.app are both able to display emoji:

![example](https://i.imgur.com/bpSPscs.png)

but alacritty isn't:

![example](https://i.imgur.com/TOLO7IX.png)

**Comment by jwilm** [comment]
Good to have a tracking issue for it :)

**Comment by topaxi** [comment]
It would be nice to be able to define a specific font for these (emoji one, for example, as not all fonts have all emojis defined), maybe even an utf8 code point range? This could also be used for something like https://github.com/ryanoasis/powerline-extra-symbols without the need to patch existing fonts.

**Comment by ton** [comment]
@topaxi, in case fontconfig is configured correctly, and alacritty supports fallback fonts (

In any case, I suspect there might be an issue related to fallback fonts (https://github.com/jwilm/alacritty/issues/45), there is no need for that kind of custom functionality.

**Comment by topaxi** [comment]
@ton, nice!

**Comment by manishrjain** [comment]
Alacritty doesn't display the icons in powerline fonts: https://github.com/powerline/fonts/tree/master/Inconsolata

I presume that bug should go here. Let me know if I should file another one.

**Comment by iv597** [comment]
On Arch Linux with Infinality and `ttf-noto-fonts-emoji-ib` I see an illegibly tiny dot-type-thing in place of emoji.

![2017-01-08-165004_944x1036_scrot](https://cloud.githubusercontent.com/assets/799439/21754275/b8e3cb4e-d5c2-11e6-88cb-fffee1033a63.png)

termite (`vte3-ng`-based) for comparison (note that while these emojis print correctly, they only use one rendering "space" because of how VTE works sans-patches. The patches make life more miserable anyway)

![2017-01-08-165213_944x1036_scrot](https://cloud.githubusercontent.com/assets/799439/21754293/f796747c-d5c2-11e6-9f42-e9140a622ff4.png)

**Comment by jwilm** [comment]
You might have some monochrome emoji on FreeType systems now

**Comment by PaulBGD** [comment]
Glad to have monochrome emojis for now:
![Image](http://i.imgur.com/TcamFhu.png)

For color Linux support it looks like you'd have to have support for _SVG in Open Type_ fonts, such as https://github.com/eosrei/emojione-color-font

**Comment by maximbaz** [comment]
SVGinOT is not the only option, it's complicated in Linux 🙂 

For example, [eosrei/emojione-color-font](https://github.com/eosrei/emojione-color-font) that you mentioned is a custom build of EmojiOne to add support for Firefox via SVGinOT technology, but the official [Ranks/emojione](https://github.com/Ranks/emojione/tree/master/extras/fonts) has [emojione-android.ttf](https://github.com/Ranks/emojione/blob/master/extras/fonts/emojione-android.ttf) which provides color emoji via CBDT/CBLC technology for all Chrome-based apps, as well as all GTK+ apps if you install a patched cairo.

By the way, the font from the official repo consists of the EmojiOne v3, which is a redesigned version of older EmojiOne v2, and the community is [working right now](https://github.com/Ranks/emojione/issues/477) to add the latest Unicode 4.0 emojis in the font.

So, in summary it would be a pity to support only SVGinOT stack.

**Comment by PaulBGD** [comment]
Huh, just hearing about CBDT/CBLC. Definitely sounds like a good solution too, as long as I can get my color emoji.

**Comment by felipesere** [comment]
Any update in this space? I'd love to get the Rocket emojis back 😄

**Comment by k0nserv** [comment]
Emoji works for me, but they are rendered semi-transparent. macOS using Fira Code

<img width="156" alt="image" src="https://user-images.githubusercontent.com/1333960/32434438-097a4956-c2d6-11e7-8307-814c29cfb045.png">

**Comment by dvergeylen** [comment]
@maximbaz, I'am afraid CBDT/CBLC isn't supported by Firefox yet. I've tried current stable and Quantum under linux, here is what I get from https://pixelambacht.nl/chromacheck/

![capture du 2017-11-06 10-44-39](https://user-images.githubusercontent.com/8804879/32434887-a6d399ba-c2df-11e7-9450-d7769eeca18e.png)

What version are you using?

**Comment by maximbaz** [comment]
I'm using Chromium on daily basis, but I have a EmojiOne font installed (which I was told is using CBDT/CBLC format) and latest official Firefox and emojis are being rendered in color:

![image](https://user-images.githubusercontent.com/1177900/32435343-ea9bbfd2-c2e0-11e7-9c9a-b0efc870c88d.png)

This website that you are referring to shows no support in both Firefox and Chromium in my case, I'm not sure if we can trust it.

**Comment by unphased** [comment]
on macOS 10.13.2 i copied 🚀 and i pasted it in Alacritty and it showed up perfectly fine.

**Comment by sodiumjoe** [comment]
i believe this was fixed in https://github.com/jwilm/alacritty/pull/956

**Comment by maximbaz** [comment]
Unfortunately only for Mac, the issue is still valid for Linux.

**Comment by alexanderadam** [comment]
@maximbaz do you know whether there is anything required to have colored emojis in sakura?

**Comment by maximbaz** [comment]
No specific configuration in sakura itself (it works with the default config), however, fontconfig must be properly configured to have colorful emojis in all linux apps (sadly, except alacritty). If you use ArchLinux, try [aur/ttf-emojione](https://aur.archlinux.org/packages/ttf-emojione/) which I maintain, otherwise here's the relevant configuration - [70-emojione-color.conf](https://aur.archlinux.org/cgit/aur.git/tree/70-emojione-color.conf?h=ttf-emojione) - which you should save as `~/.config/fontconfig/conf.d/70-emojione-color.conf` and then refresh the cache with `$ fc-cache -f; sudo fc-cache -f`.

**Comment by ngocphamm** [comment]
Am I crazy but on Mac, Alacritty is still not displaying emoji the same way as other apps?

Alacritty
<img width="301" alt="2018-06-10 at 12 34 am" src="https://user-images.githubusercontent.com/395123/41198352-760dd6b8-6c46-11e8-8317-e55a6daf03d3.png">


iTerm2
<img width="312" alt="2018-06-10 at 12 35 am" src="https://user-images.githubusercontent.com/395123/41198348-71b984c2-6c46-11e8-85eb-57a4de59a540.png">


Sublime Text
<img width="329" alt="2018-06-10 at 12 36 am" src="https://user-images.githubusercontent.com/395123/41198347-6d841cfa-6c46-11e8-8c00-e0d09a4e2ece.png">

**Comment by chrisduerr** [comment]
@ngocphamm This is probably a font configuration thing. This might be fixed once work on #957 starts.

**Comment by bmisiak** [comment]
![image](https://user-images.githubusercontent.com/2332213/52909511-5cee6800-323e-11e9-9aee-10079e082a3e.png)
On Windows 1809 (ConPTY backend), emojis don't show up on the default config.

**Comment by saifali96** [comment]
Guys, I'm sorry if the question sounds uninformed, does Alacritty support emojis on Windows at all? Even with the new ConPty interface?

**Comment by zachliu** [comment]
humble linux user awaits :crossed_fingers: 

alacritty:
![alacritty](https://user-images.githubusercontent.com/14293802/61924156-96857680-af34-11e9-9f0e-f5973dac8911.png)

terminal:
![terminal](https://user-images.githubusercontent.com/14293802/61924157-98e7d080-af34-11e9-8f0f-46469f449477.png)

i am alacritty 0.3.3 on Mint 19.1 tessa

**Comment by chrisduerr** [comment]
@tkolleh That looks like some other font might provide this emoji before your actual emoji font. You could try to change the default font and check if that fixes it.

**Comment by tkolleh** [comment]
@chrisduerr Thank you. That looks to be the challenge. Now to figure out how to use the font and display the emoji. Similiar behavior is demonstrated in markdown by escaping the emoji.

`local LINEINDICATOR="⚡️"`
<br>
local LINEINDICATOR="⚡️"

**Comment by zachliu** [comment]
somehow it doesn't work, my :rocket: is still in the same colorless shape :sob: 
did i miss anything? any extra setups to make this work?

**Comment by kchibisov** [comment]
You should properly configure your font fallback and use emojis compatible with freetype (e.g. Noto Color Emoji).

**Comment by dvergeylen** [comment]
> You should properly configure your font fallback and use emojis compatible with freetype (e.g. Noto Color Emoji).

How do we do that (asking for a friend)? 😇

**Comment by kchibisov** [comment]
https://github.com/eosrei/twemoji-color-font/blob/master/linux/fontconfig/56-twemoji-color.conf Try this as a pattern, but use Noto Color Emoji instead of twitter ones. Put it in .config/fontconfig/fonts.conf idk. Use this thing to debug 🤔

**Comment by cole-h** [comment]
Also see my comment here: https://github.com/jwilm/alacritty/pull/3011#issuecomment-563475481

**Comment by lovesegfault** [comment]
@cole-h I've used your code but have been unable to get color emojis to show up normally.

**Comment by cole-h** [comment]
Do you have Noto Color Emoji installed? If so, what does `fc-match 'Noto Color Emoji'` print?

If you're trying an emoji font other than Noto Color, you'll probably have to replace the name at every occurrence inside that XML-structure conf file.

**Comment by lovesegfault** [comment]
For reference:
```
❯ fc-match 'Noto Color Emoji'
NotoColorEmoji.ttf: "Noto Color Emoji" "Regular"
```

will ping you on IRC.

**Comment by wimstefan** [comment]
I have to chime in and report that I have the same issue than @lovesegfault and @zachliu  no matter which of the suggested fallback configurations I use :-/

**Comment by chrisduerr** [comment]
This is probably more related to #1560 than colored emojis. Obviously if the first matching emoji font is not colored, you won't get colored emojis. This specifically fixes the problem where no emojis are rendered at all, because the colored emoji is picked but can't be rendered.

For people running into fontconfig fallback issues, it would be extremely helpful if they could provide feedback if the PR https://github.com/jwilm/alacritty/pull/2935 helps with this.

**Comment by Lythenas** [comment]
I can confirm that most emojis are displayed with #2935. Some of the basic face emojis are not colored (maybe they are included in my font). I'm using Iosevka Nerd Font on Arch.

On v0.4.0 it doesn't display any emojis. Also xterm doesn't display most emojis but Terminator works fine. I think my fontconfig is correct. Can't be sure though. I followed https://github.com/jwilm/alacritty/pull/3011#issuecomment-563475481

**Comment by wezm** [comment]
Yep I can confirm it works too. I'm using the [ttf-joypixels package on Arch Linux](https://www.archlinux.org/packages/community/any/ttf-joypixels/). Note that you have to symlink the config after installation to configure fontconfig to use it (`sudo ln -sf /etc/fonts/conf.avail/75-joypixels.conf /etc/fonts/conf.d/75-joypixels.conf`).

Also it only seems to support single emoji, not emoji ZWJ sequences.

![Screenshot from 2019-12-10 10-30-02](https://user-images.githubusercontent.com/21787/70755631-785ace80-1d8e-11ea-9755-f951d7b17775.png)

**Comment by chrisduerr** [comment]
> Also it only seems to support single emoji, not emoji ZWJ sequences.

That is correct and there's no intention to support that at the moment.

**Comment by Findarato** [comment]
Just adding in this still is not working with all the above fixes.  Not sure if this was closed because it was fixed or not.

**Comment by kchibisov** [comment]
>Just adding in this still is not working with all the above fixes. Not sure if this was closed because it was fixed or not.

It was closed because it was implemented as you can see in the reason of close on a github. However you need an emoji font that freetype can understand.

To make things clear.

We're not gating emojis like most apps do. So if you see them monochrome some fonts could come before, etc. The better handling of fallback will be added with #957 and I also has some notes laying around about fontconfig and how its fallback can behave in a weird way, so there's a change that we land something before rfc being implemented.

One more reason of monochrome emojis is a font that freetype couldn't understand, like Twitter Color Emoji with SVG in Open Type.

**Comment by abelsromero** [comment]
> It was closed because it was implemented as you can see in the reason of close on a github. However you need an emoji font that freetype can understand.

I am running from the master branch with NotoColorEmoji and still not working. I see the PR #3163 mentions fallback configuration but the example configuration in the repo does not show how and every attempt to try different syntaxes fails. If it is indeed fixed, how then is this configured, all other applications work fine for me.

**Comment by jeremie-H** [comment]
I've got same here : https://github.com/alacritty/alacritty/issues/3294
time to time, i try a git pull here and hope ^^
I've tried anything, but i havent found a workaround :(

i'm back to gnome-terminal since

**Comment by jeremie-H** [comment]
> > It was closed because it was implemented as you can see in the reason of close on a github. However you need an emoji font that freetype can understand.
> 
> I am running from the master branch with NotoColorEmoji and still not working. I see the PR #3163 mentions fallback configuration but the example configuration in the repo does not show how and every attempt to try different syntaxes fails. If it is indeed fixed, how then is this configured, all other applications work fine for me.

I've got same here : https://github.com/alacritty/alacritty/issues/3294
time to time, i try a git pull here and hope ^^
I've tried anything, but i havent found a workaround :(

i'm back to gnome-terminal since

**Comment by jeremie-H** [comment]
Hello,
I think we should reopen this issue, i've tried an easy step to reproduce this still complex emoji issue. It doesn't display colored emoji :

_easy step to reproduce_  

- download ubuntu 20.04 iso
- install fresh ubuntu 20.04, default desktop
- follow install instructions :
```
sudo add-apt-repository ppa:mmstick76/alacritty
sudo apt install alacritty
```
- copy/paste in alacritty the following command : 
`echo 🐵`   

### Result
![Capture d’écran de 2020-04-27 19-24-21](https://user-images.githubusercontent.com/11060749/80401517-bf200f80-88bc-11ea-9b4c-20ab6b915804.png)
It doesnt work :-(

**Comment by kchibisov** [comment]
Post me the output of `fc-match -s YOUR_FONT_IN_ALACRITTY` from alacritty v0.4.2, if the version isn't v0.4.2 (you can check that by `alacritty --version`)? If it's not v0.4.2, make sure to use v0.4.2.

**Comment by jeremie-H** [comment]
ok, you're right, it's the default install 0.4.1 in **repository up to date** ...
i can try with a fresh v0.5 if you want ?
but i need to install rustup, cargo and so on ... i just want to show that existing issue is still alive in a fresh install

**Comment by jeremie-H** [comment]
and i havent specify **ANY** YOUR_FONT_IN_ALACRITTY.
All is by default install

**Comment by kchibisov** [comment]
then query by `fc-match -s monospace`.

**Comment by kchibisov** [comment]
v0.4.2 is fine, v0.5.0 isn't released yet, and is a dev version.

**Comment by jeremie-H** [comment]
0.4.2 isnt fine more than 0.4.1

![Capture d’écran de 2020-04-27 19-36-13](https://user-images.githubusercontent.com/11060749/80402639-78331980-88be-11ea-958f-e56c662d5587.png)

**Comment by jeremie-H** [comment]
steps are easy to reproduce, all configuration is **default**, a virtualbox with fresh ubuntu install

**Comment by kchibisov** [comment]
As I said, pst the output of `fc-match -s monospace`.

**Comment by jeremie-H** [comment]
As I said ... you've got an **easy steps** to reproduce. It couldn't be simpler, It doesn't work **out of the box** on a fresh install.
it should be sufficient to reopen this issue.

**Comment by jeremie-H** [comment]
if it could make you happy, here your output of `fc-match -s monospace`
```
DejaVuSansMono.ttf: "DejaVu Sans Mono" "Book"
DejaVuSansMono-Bold.ttf: "DejaVu Sans Mono" "Bold"
NimbusMonoPS-Regular.otf: "Nimbus Mono PS" "Regular"
TlwgTypo.ttf: "Tlwg Typo" "Regular"
TlwgTypo-Bold.ttf: "Tlwg Typo" "Bold"
NotoSansCJK-Regular.ttc: "Noto Sans Mono CJK JP" "Regular"
KhmerOSsys.ttf: "Khmer OS System" "Regular"
mitra.ttf: "Mitra Mono" "Regular"
Lohit-Bengali.ttf: "Lohit Bengali" "Regular"
Lohit-Gujarati.ttf: "Lohit Gujarati" "Regular"
Lohit-Tamil.ttf: "Lohit Tamil" "Regular"
Meera-Regular.ttf: "Meera" "Regular"
Lohit-Kannada.ttf: "Lohit Kannada" "Regular"
Lohit-Telugu.ttf: "Lohit Telugu" "Regular"
lklug.ttf: "LKLUG" "Regular"
FreeMono.ttf: "FreeMono" "Regular"
FreeSans.ttf: "FreeSans" "Regular"
FreeSerif.ttf: "FreeSerif" "Regular"
opens___.ttf: "OpenSymbol" "Regular"
Norasi.ttf: "Norasi" "Regular"
Rachana-Regular.ttf: "Rachana" "Regular"
KacstOne.ttf: "KacstOne" "Regular"
Laksaman.ttf: "Laksaman" "Regular"
Purisa.ttf: "Purisa" "Regular"
Padauk-Regular.ttf: "Padauk" "Regular"
Sawasdee.ttf: "Sawasdee" "Regular"
Lohit-Devanagari.ttf: "Lohit Devanagari" "Regular"
DejaVuSans.ttf: "DejaVu Sans" "Book"
DejaVuSerif.ttf: "DejaVu Serif" "Book"
LiberationMono-Regular.ttf: "Liberation Mono" "Regular"
AbyssinicaSIL-Regular.ttf: "Abyssinica SIL" "Regular"
D050000L.otf: "D050000L" "Regular"
TibetanMachineUni.ttf: "Tibetan Machine Uni" "Regular"
Manjari-Regular.otf: "Manjari" "Regular"
Pothana2000.ttf: "Pothana2000" "Regular"
chandas1-2.ttf: "Chandas" "Regular"
Pagul.ttf: "Pagul" "猀 愀瀀瀀氀"
mry_KacstQurn.ttf: "mry_KacstQurn" "Regular"
Rasa-Regular.ttf: "Rasa" "Regular"
Waree.ttf: "Waree" "Regular"
Phetsarath_OT.ttf: "Phetsarath OT" "Phetsarath OT"
DroidSansFallbackFull.ttf: "Droid Sans Fallback" "Regular"
Ubuntu-C.ttf: "Ubuntu Condensed" "Regular"
Ubuntu-R.ttf: "Ubuntu" "Regular"
StandardSymbolsPS.t1: "Standard Symbols PS" "Regular"
Samyak-Devanagari.ttf: "Samyak Devanagari" "Regular"
Samyak-Gujarati.ttf: "Samyak Gujarati" "Regular"
padmaa.ttf: "padmaa" "Normal"
Rekha.ttf: "Rekha" "Medium"
utkal.ttf: "ori1Uni" "Medium"
padmaa-Bold.1.1.ttf: "padmaa-Bold.1.1" "Bold.1.1"
LiberationSerif-Italic.ttf: "Liberation Serif" "Italic"
NotoColorEmoji.ttf: "Noto Color Emoji" "Regular"
```

**Comment by kchibisov** [comment]
Yeah, as it was expected, your `NotoColorEmoji` font is the last one, so I guess you're getting this monochrome emoji from some other of these bazillion fonts you have. The only solution is to define your own fontconfig rules. On a lot of systems they ship rules, some folks define their own fontconfig rules, some apps hard code emoji fonts.

You'll need similar rules from https://github.com/eosrei/twemoji-color-font/blob/master/linux/fontconfig/56-twemoji-color.conf , but for Noto Color Emoji and your font. Then you should place them in ~/.config/fontconfig/fonts.conf or add them globaly to the system fonts config.

This issue is purely about rendering colored emoji glyphs if you fallback to a colored emoji font during fallback, which is why it was closed, since we can render them now. If you end up falling back to a monochrome emoji font.

And this is also example on how to configure them https://github.com/alacritty/alacritty/issues/153#issuecomment-565224608 and you can see them being rendered.

So please, don't ping this issue if your system provides not desired font configuration by default or don't ship required fontconfig rules.

>As I said ... you've got an easy steps to reproduce. It couldn't be simpler, It doesn't work out of the box on a fresh install.
it should be sufficient to reopen this issue.

I don't have time to setup vm with ubuntu to prove again, that you should define fontconfig rules. There're a lot of references on how to that and is quite a common problem.

**Comment by abelsromero** [comment]
Thanks to the points rised by @kchibisow I could delve deeper and fix it with this instructions https://forum.manjaro.org/t/tutorial-how-to-enable-system-wide-color-emoji-support/35188. For whoever needs it, check the comment below and not the main post for a user only configuration.

**Comment by dvergeylen** [comment]
For Debian / Ubuntu users, paste this in `~/.config/fontconfig/fonts.conf`:
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
 <alias>
  <family>monospace</family>
  <prefer>
    <family>DejaVu Sans Mono</family>
    <family>Noto Color Emoji</family>
    <family>Noto Emoji</family>
   </prefer>
 </alias>
</fontconfig>
```
This will render MOST of the emojis in color.

Before:
![01_before](https://user-images.githubusercontent.com/8804879/80486367-55a30e00-895b-11ea-9ece-97009e965ab1.png)

After:
![02_after](https://user-images.githubusercontent.com/8804879/80486369-563ba480-895b-11ea-8749-ee627827b435.png)

Testing line: `echo "♥👍😍🙊🥱🥺✨✅🎊🏄"`

**Comment by arooni** [comment]
I tried this from the help of someone in the #alacritty irc channel today but couldn't get it working

```
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
<alias>
<family>monospace</family>
<prefer>
<family>Ubuntu Mono derivative Powerline</family>
<family>Noto Color Emoji</family>
<family>Noto Emoji</family>
</prefer>
</alias>
</fontconfig>
```

**Comment by abelsalgadoromeroWD** [comment]
@arooni that worked for me. Obvious suggestions but...
1. Make sure Noto Color Emoji is installed correctly.
![image](https://user-images.githubusercontent.com/56251795/80568445-4c677f00-89f7-11ea-90c3-51337088a33f.png)
2. Ensure your rule is under `$HOME/.config/fontconfig` and is called `fonts.conf`.

**Comment by sQVe** [comment]
I managed to set this up after some minor struggles. The following **should hopefully help** others to get **everything setup properly**:
1. Start with installing **Noto Color Emoji**. On Arch: 
```sh
% yay -S noto-fonts-emoji
```
2. Force a font rescan with:
```sh
% fc-cache --really-force
```
3. Ensure that the font is available by running:
```sh
% fc-list | grep -i "emoji"
/usr/share/fonts/noto/NotoColorEmoji.ttf: Noto Color Emoji:style=Regular
```
4. Edit the user `fonts.conf` include our *emoji* font above:
```sh
% $EDITOR $XDG_CONFIG_HOME/fontconfig/fonts.conf
```
5. Either add or edit your `monospace` family entry, for example:
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
  <alias>
    <family>monospace</family>
    <prefer>
      <family>Cousine</family> <!-- Your preferred monospace font -->
      <family>Noto Color Emoji</family> <!-- Your preferred emoji font -->
     </prefer>
  </alias>
</fontconfig>
```
6. Force rescan again if you made any changes.
```sh
% fc-cache --really-force
```

7. Ensure that you have set `monospace` as your font in `alacritty.yml`: 
```sh
% cat $XDG_CONFIG_HOME/alacritty/alacritty.yml                   ~
...
font:
  normal:
    family: monospace
    style: Regular
  bold:
    family: monospace
    style: Bold
  italic:
    family: monospace
    style: Italic
  size: 18
  offset:
    x: 0
    y: 0
  glyph_offset:
    x: 0
    y: 0
...
```
8. Open a new `alacritty` terminal.
9. Test your success with `echo "♥👍😍🙊🥱🥺✨✅🎊🏄"`.

---

**Note**: You can create any family alias you want in your `font.conf`. Instead of using `monospace` you could create a family called `foobar`. For example:
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
  <alias>
    <family>foobar</family>
    <prefer>
      <family>Cousine</family> <!-- Your preferred monospace font -->
      <family>Noto Color Emoji</family> <!-- Your preferred emoji font -->
     </prefer>
  </alias>
</fontconfig>
```

**Comment by zachliu** [comment]
i'm on `linux mint 19.3` and using `alacritty 0.5.0-dev (e009869)`. this is my `zsh` (v5.7.1) and `bash` (v4.4.20):
![Screenshot from 2020-06-04 14-15-17](https://user-images.githubusercontent.com/14293802/83795740-ead9a680-a66d-11ea-8dd6-02d0f652c82c.png)

don't know why it looks different with yours @dvergeylen  `¯\_(ツ)_/¯`

this is on my chrome
![screenshot-github com-2020 06 04-14_16_47](https://user-images.githubusercontent.com/14293802/83795850-16f52780-a66e-11ea-9740-c70a8418ddff.png)

this is my `.config/fontconfig/font.conf`
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>

    <!-- Add generic family. -->
    <match target="pattern">
        <test qual="any" name="family"><string>emoji</string></test>
        <edit name="family" mode="assign" binding="same"><string>Noto Color Emoji</string></edit>
    </match>

    <!-- This adds Noto Color Emoji as a final fallback font for the default font families. -->
    <match target="pattern">
        <test name="family"><string>sans</string></test>
        <edit name="family" mode="append"><string>Noto Color Emoji</string></edit>
    </match>

    <match target="pattern">
        <test name="family"><string>serif</string></test>
        <edit name="family" mode="append"><string>Noto Color Emoji</string></edit>
    </match>

    <match target="pattern">
        <test name="family"><string>sans-serif</string></test>
        <edit name="family" mode="append"><string>Noto Color Emoji</string></edit>
    </match>

    <match target="pattern">
        <test name="family"><string>monospace</string></test>
        <edit name="family" mode="append"><string>Noto Color Emoji</string></edit>
    </match>

    <!-- Block Symbola from the list of fallback fonts. -->
    <selectfont>
        <rejectfont>
            <pattern>
                <patelt name="family">
                    <string>Symbola</string>
                </patelt>
            </pattern>
        </rejectfont>
    </selectfont>

</fontconfig>
```
this is my `~/.font/`
```bash
.
├── AppleColorEmoji.ttf
├── Cousine
│   ├── Cousine Bold Italic Nerd Font Complete Mono.ttf
│   ├── Cousine Bold Italic Nerd Font Complete Mono Windows Compatible.ttf
│   ├── Cousine Bold Italic Nerd Font Complete.ttf
│   ├── Cousine Bold Italic Nerd Font Complete Windows Compatible.ttf
│   ├── Cousine Bold Nerd Font Complete Mono.ttf
│   ├── Cousine Bold Nerd Font Complete Mono Windows Compatible.ttf
│   ├── Cousine Bold Nerd Font Complete.ttf
│   ├── Cousine Bold Nerd Font Complete Windows Compatible.ttf
│   ├── Cousine Italic Nerd Font Complete Mono.ttf
│   ├── Cousine Italic Nerd Font Complete Mono Windows Compatible.ttf
│   ├── Cousine Italic Nerd Font Complete.ttf
│   ├── Cousine Italic Nerd Font Complete Windows Compatible.ttf
│   ├── Cousine Regular Nerd Font Complete Mono.ttf
│   ├── Cousine Regular Nerd Font Complete Mono Windows Compatible.ttf
│   ├── Cousine Regular Nerd Font Complete.ttf
│   └── Cousine Regular Nerd Font Complete Windows Compatible.ttf
├── NotoColorEmoji.ttf
├── NotoEmoji-Regular.ttf
└── TwitterColorEmoji-SVGinOT.ttf
```

**Comment by manofthelionarmy** [comment]
> For Debian / Ubuntu users, paste this in `~/.config/fontconfig/fonts.conf`:
> 
> ```
> <?xml version="1.0"?>
> <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
> <fontconfig>
>  <alias>
>   <family>monospace</family>
>   <prefer>
>     <family>DejaVu Sans Mono</family>
>     <family>Noto Color Emoji</family>
>     <family>Noto Emoji</family>
>    </prefer>
>  </alias>
> </fontconfig>
> ```
> 
> This will render MOST of the emojis in color.
> 
> Before:
> ![01_before](https://user-images.githubusercontent.com/8804879/80486367-55a30e00-895b-11ea-9ece-97009e965ab1.png)
> 
> After:
> ![02_after](https://user-images.githubusercontent.com/8804879/80486369-563ba480-895b-11ea-8749-ee627827b435.png)
> 
> Testing line: `echo "♥👍😍🙊🥱🥺✨✅🎊🏄"`
> 
> **Edit:** You can of course use `Noto Mono` instead of `DejaVu Sans Mono` as proposed by **abelsromero**'s link, but this renders differently than the latter (default Debian/Ubuntu terminal's font?):
> 
> ![](https://user-images.githubusercontent.com/8804879/80487138-9cddce80-895c-11ea-8d1b-59add0c395ac.png)

if anyone is using Ubuntu 20.04, according to this: https://manpages.ubuntu.com/manpages/focal/en/man5/fonts-conf.5.html,
make sure to export XDG_CONFIG_HOME in your .bashrc, as such:
```
export XDG_CONFIG_HOME=$HOME/.config
```
after wards, in your terminal, run

```
source ~/.bashrc
fc-cache -rv
```
Make sure your fonts are installed in` ~/.local/share/fonts/` as well.

**Comment by eggbean** [comment]
This solution works for me and only works on a specific font, so is unlikely to make any unwanted changes elsewhere. I am using Pop!_OS 21.04 (Debian/Ubuntu based). I am using Hack as my terminal font:

`~/.config/alacritty.yml`
```
# Font configuration
  font:
    # Normal (roman) font face
    normal:
      family: "Hack Nerd Font"
      style: Regular

    # Bold font face
    bold:
      style: Bold

    # Italic font face
    italic:
~     style: Italic

    # Bold italic font face
    bold_italic:
~     style: Bold Italic

    # Point size
    size: 13.0

  draw_bold_text_with_bright_colors: true
```
`~/.config/fontconfig/fonts.conf`
```
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
    <match target="font">
        <!-- If the requested font is Hack Nerd Font -->
        <test name="family" compare="eq">
            <string>Hack Nerd Font</string>
        </test>
        <!-- Replace the entire match list with Hack Nerd Font alone -->
        <edit name="family" mode="assign_replace">
            <string>Hack Nerd Font</string>
        </edit>
        <!-- Followed by Noto Color Emoji -->
        <edit name="family" mode="prepend_first">
            <string>Noto Color Emoji</string>
        </edit>
    </match>
</fontconfig>
```

**Comment by colasgabens** [comment]
Thanks a lot but i fix it a long time ago and yes i did what you just sent
me thanks a lot

**Comment by eggbean** [comment]
Okay, just thought it could be useful for other people.

**Comment by colasgabens** [comment]
Thanks for the tip!

**Comment by dym-sh** [comment]
ok, so what finally fixed it for me is to have only `Noto Color Emoji` font in every family — AND NOTHING ELSE!
other fonts like `Twitter Color Emoji` will fall back to outlines

![emoji fix](https://user-images.githubusercontent.com/59403389/152649735-55708d37-d6fe-473a-adab-ef2ff1bb8dc5.png)

> `.fonts.conf`

```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
  <alias>
   <family>sans-serif</family>
   <prefer>
     <family>Noto Color Emoji</family>
   </prefer>
  </alias>

  <alias>
   <family>serif</family>
   <prefer>
     <family>Noto Color Emoji</family>
   </prefer>
  </alias>

  <alias>
    <family>monospace</family>
    <prefer>
      <family>Noto Color Emoji</family>
    </prefer>
  </alias>

</fontconfig>
```

**Comment by illia-danko** [comment]
The issued has appeared to me recently on alacritty 0.11.0. Fixed by installing two package on arch: noto-fonts-emoji, noto-color-emoji-fontconfig

**Comment by 4127157** [comment]
@illia-danko could you share your `font.conf` file content? 
`~/.config/fontconfig/font.conf`

**Comment by illia-danko** [comment]
> @illia-danko could you share your `font.conf` file content? `~/.config/fontconfig/font.conf`

I rather put it under global settings /etc/fonts/local.conf. But it looks like nothing related to alacritty itself:
https://github.com/illia-danko/dotfiles/blob/09dc10659e51b81038d10ed3dc08d4135694e22e/root/etc/fonts/local.conf#L1

**Comment by rafay99-epic** [comment]
# Question:
Hey, does anyone have any idea what the cause of these icons not showing is?or rendering, because I have no idea.
The icons are rendering any colors. Can you explain what is happening here 

## Images
![image](https://github.com/alacritty/alacritty/assets/82662797/5cac6336-4b42-41a7-b0ad-5ea6c5b12cd0)

![image](https://github.com/alacritty/alacritty/assets/82662797/65e58f8b-4c25-4c01-ac81-9d55dac18f75)

**Comment by ahrasheed** [comment]
Emojis are broken again when building from master on Ubuntu 22.04. This occurred after commit  [#48c088](https://github.com/alacritty/alacritty/commit/48c088a50c977dfaf2a5a7052a5ddd39071e6063). This problem occurs independent on font config and can be seen in the screenshots below. Both versions use exactly the same config for alacritty and for fonts. 

**Working in commit [#a77f77c](https://github.com/alacritty/alacritty/commit/a77f77c48fca298caab3a4834b2d7ab1a98cae88)**
![Screenshot from 2024-05-25 18-38-29](https://github.com/alacritty/alacritty/assets/3121716/d038df6f-9763-4732-a05f-bab68b8e85e2)
**Broken in commit [#48c088](https://github.com/alacritty/alacritty/commit/48c088a50c977dfaf2a5a7052a5ddd39071e6063)**
![Screenshot from 2024-05-25 18-38-43](https://github.com/alacritty/alacritty/assets/3121716/245cb643-9fd8-4046-b5e8-bc161469819d)

**Comment by anamyk** [comment]
I have the same problem with `alacritty 0.15.0-dev`

**Comment by rafay99-epic** [comment]
No, matter, which version I used, I compiled from source, or via Package Manager, the issue of displaying emojis does not go away.

**Comment by j-lakeman** [comment]
What fixed it for me on both Debian and Alpine was following this: https://wiki.alpinelinux.org/wiki/Emojis
Similar to previous comments, but kind of a rather “official” resource.

**Comment by rafay99-epic** [comment]
> What fixed it for me on both Debian and Alpine was following this: https://wiki.alpinelinux.org/wiki/Emojis Similar to previous comments, but a rather “official” resource.

Thanks for the help, But I don't use Alacrity any more, I have installed on my system but not using it, kitty is my default terminal for now

**Comment by NeoSahadeo** [comment]
> What fixed it for me on both Debian and Alpine was following this: https://wiki.alpinelinux.org/wiki/Emojis Similar to previous comments, but a rather “official” resource.

This worked for me. I'm on void
